app-info/generate-eos-extra-appstream

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/python3
+# Copyright © 2020 Endless OS Foundation LLC
+#
+# Generator for external appstream XML
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA
+
+import argparse
+import glob
+import os
+from xml.etree import ElementTree
+
+
+def main():
+    # Provide some basic --help output
+    parser = argparse.ArgumentParser(
+        description='Generate eos-extra.xml appstream from metainfo files',
+    )
+    parser.parse_args()
+
+    appinfo_dir = os.path.dirname(__file__)
+    metainfo_dir = os.path.join(appinfo_dir, 'metainfo')
+
+    catalog = ElementTree.Element("components", version="0.8", origin="flathub")
```

Comment:
Seeing this move between the two scripts jogged a faint memory of there being some good reason why we claim the origin for these components is `flathub` when a good number of them are from `eos-apps`. And I felt like we ought to have a comment justifying this lie.

d85bd3c32bada70eafb7ac3cc98b20b38ce435bd changed it from `origin="eos-extra-app-info"` to `origin="flathub"` without explanation. Clicking through to the linked ticket, [I asked the same question back then in 2019](https://phabricator.endlessm.com/T26507#741335). Lower down, I found a [link](https://phabricator.endlessm.com/T26507#741741) to a [paste](https://phabricator.endlessm.com/P664) with the following from Richard Hughes on IRC:

```
11:12 <hughsie> i don't think the origin name is important for external appstream
11:12 <hughsie> e.g. just leave it blank
```

Evidently the `origin` doesn't actually have to match because marking `eos-apps` apps as noteworthy works fine with this. And actually the appstream for `eos-apps` and `flathub` both sets:

```xml
<components version="0.8" origin="flatpak">
```

So maybe it really doesn't matter and the original was right?

Reply:
Thinking on this again, I think we should change it to `eos` or `eos-extra` like you say. I don't know if gnome-software will retain the origin in the installed PWA, but I think it would be good to fix it now before we start deploying this appstream.

Reply:
I added a commit to change it to `eos-extra` as the spec suggests it should match the catalog basename. I hope there are no regressions from this.